src/components/usuarios/UsuarioForm.jsx

```diff
@@ -0,0 +1,268 @@
+import React, { Component } from 'react';
+import { firebaseConnect } from 'react-redux-firebase';
+import { withRouter } from 'react-router-dom';
+
+import axios from 'axios';
+
+import randomString from 'random-string';
+
+@firebaseConnect()
+@withRouter
+class UsuarioForm extends Component {
+
+  state = {
+    active: false,
+    role: '',
+    name: '',
+    email: '',
+    displayName: '',
+    password: '',
+    error: {
+      message: '',
+    },
+    isLoading: false,
+  }
+
+  constructor(props) {
+    super(props);
+
+    // If component recieves usuario as prop we merge it with initial state (used for editing)
+    this.state = { ...this.state, ...props.usuario };
+  }
+
+  addUsuario() {
+    const { active, role, name, email, displayName } = this.state;
+    let { password } = this.state;
+
+    // create context references for callback
+    const _this = this;
+    const { firebase } = this.props;
+
+    // generate random password if empty
+    if (password === '' || password === undefined) {
+      password = randomString({length: 10});
+    }
+
+    // create user function url
+    const createUser = 'https://us-central1-igv-andamiaje-co.cloudfunctions.net/createUser';
```

Comment:
should this value be passed as a Global or something?

Reply:
I mean the base string before the function

Reply:
yes imported as constant